### DISCUSSION TITLE
Server Hook Returning incorrect time

### DISCUSSION THREAD

**Discussion by Mashiane** [pocketbase/pocketbase#5381] [Q&A] [answered]
Hi there

I have a hook that is returning the server date and time. For my app, end users can change the date so I wanted to use a fixed time of where the app is installed.

![image](https://github.com/user-attachments/assets/a6f78b13-7411-4bda-b8b8-35ea0ca2875c)

I have noted that it returns the correct time mostly and also returns the incorrect time. For example, one users laptop time is 16:03 however the time returned by the server is 07:01.

Any ideas?

**Comment by ganigeorgiev** [comment]
_Please don't post code as screenshot. It is hard to read and copy in case it needs to be run locally._

It is not clear what do you mean by _"returns the correct time mostly and also returns the incorrect time"_. Testing locally the goja `Date` constructor works correctly for me.

goja also has several [`Date` related tests](https://github.com/dop251/goja/blob/master/date_test.go) to ensure that the constructor is working as expected.

> ... one users laptop time is 16:03 however the time returned by the server is 07:01.

`new Date()` create a new date in the device's **local time**.

If the "user laptop" and the "server" are not in the same timezone then it is possible the two dates to differ if you are not taking into account the timezone differences when comparing them.

Additionally, note that even if they were using the same timezone there is no guarantee that the server time will match exactly with that on the user device (_how do you even comparing them? do you take into account the time between the network request and the check?_).

If you think that there is an issue in PocketBase feel free to provide clear reproducible steps and I'll try to investigate it but as of now the post doesn't really make sense to me.

**Comment by Mashiane** [comment]
Thank you for the clarity. I am attaching here the file with the hook. This is residing on the server where my pocketbase app is running from.

[servertime.pb.js.zip](https://github.com/user-attachments/files/16715258/servertime.pb.js.zip)

I have an app published there on the pb_public folder. So I am calling this hook with fetch api from my app to get the time from the same server. I am trying to avoid reading the date time from the end user device this way. In all my tests this worked well, even with my device and laptop. Now one end user is saying his time is incorrect, 16:03 (on laptop) vs 07:01 (from this hook) showing on the app. My app time is correct, so the hook is accurately reflecting.

That's where I got buffled as to what could be happening. It might not be a pocketbase issue for sure, it might just be his windows device.

**Reply by ganigeorgiev** [reply, marked as answer]
Sorry I don't have a VM setup at the moment and can't download and open random zip files.

Note that code blocks can be pasted as markdown by wrapping it with 3-backticks.

Again the goja `Date` constructor works correctly for me and your issue is most likely related to your client-side code. Make sure that you are taking the local time zones in consideration. Other than this I don't think I can help you further.

**Reply by Mashiane** [reply]
Thank you so much. Been a long day, I recognize that my mind switched off for a while, someone pointed me to what the problem was actually.

The end user was using the development server vs the public one. Phew. I have pointed them to the right server to use. Everything is perfect, sorry for wasting your time, unintentional.
